### PR TITLE
sql/parser: Overhaul error handling during parsing of ColumnTableDefs

### PR DESCRIPTION
**Pull request by nvb**
An exploration into why multiple `CHECK` constraints with the same name for the same column weren't throwing an error led to the discovery that `newColumnTableDef` was pretty buggy and that we did not even support multiple `CHECK` constraints for the same column. Most fields were simply replaced by later qualifications, without concern for duplicate specifications or conflicting conditions. This series of changes:
- Cleans up the function by removing unnecessary string checks
- Disallows multiple default values on a single column
- Disallows conflicting NULL/NOT NULL declarations on a single column
- Permits multiple CHECK CONSTRAINTS on the same column (which in turn fixes my initial question)
- Disallows multiple FK References and Families on a single column

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9868)

### CODE REVIEW

**Comment by tamird** [comment]
:lgtm_strong: 

---

Reviewed 1 of 1 files at r1, 5 of 5 files at r2, 2 of 2 files at r3, 7 of 7 files at r4, 5 of 5 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9868#-:-KTnRZ_pUmOoamyBc1l9:b-uwkowd)*

File: sql/testdata/table [hunk 1]
```diff
@@ -282,6 +282,12 @@ CREATE TABLE test.dupe_named_constraints (
   INDEX (name)
 )
 
+statement error duplicate constraint name: "one"
+CREATE TABLE test.dupe_named_constraints (
+  id        INT CONSTRAINT pk PRIMARY KEY,
+  title     VARCHAR CONSTRAINT one CHECK (1>1) CONSTRAINT one CHECK (1<1)
+)
+
```
Comment (dt) [inline sql/testdata/table]: Might be worth including tests for the other usage errors now explicitly disallowed (multiple defaults, references, etc).

**Review by dt — APPROVED** [review]
👍

File: sql/testdata/table (hunk 1 above)
Comment (nvanbenschoten) [inline sql/testdata/table]: These are all tested at the parser level in `parse_test.go`.

**Comment by nvanbenschoten** [comment]
TFTRs!

**Pull request merged**